sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -131,6 +161,54 @@ void receiveMessageAutoComplete(MessagingEntityType entityType, boolean isSessio
         messagesPending.decrementAndGet();
     }
 
+    /**
+     * Verifies that we can send messaged with some delay and receive a message without receiver dying.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveSlowMultipleMessageAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        int count = 3;
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        Duration delayPublishing = Duration.ofMillis(1000);
+
+        final String messageId = UUID.randomUUID().toString();
+        AtomicReference<List<ServiceBusReceivedMessage>> receivedMessages = new AtomicReference<>();
+        receivedMessages.set(new ArrayList<>());
+
+        Flux.just(
+            getMessage(messageId, isSessionEnabled),
+            getMessage(messageId, isSessionEnabled),
+            getMessage(messageId, isSessionEnabled)
+        )
+            .delayElements(delayPublishing)
+            .map(message -> sender.send(message).subscribe())
+            .subscribe();
+
+        // Assert & Act
+        StepVerifier.create(receiver.receive())
+            .assertNext(receivedMessage -> {
+                assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+                receivedMessages.get().add(receivedMessage);
+            })
+            .assertNext(receivedMessage -> {
+                assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+                receivedMessages.get().add(receivedMessage);
+            })
+            .assertNext(receivedMessage -> {
+                assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+                receivedMessages.get().add(receivedMessage);
+            })
+            .thenCancel()
+            .verify();
+
+        for (int index = 0; index < count; ++index) {
+            messagesPending.decrementAndGet();
```

Comment:
You're receiving in PEEK_MODE, these messages will still be in the messaging entity, so it'll be missed during clean up. Messages pending is never incremented, this is going to be negative.